apps/prairielearn/src/components/Navbar.html.ts

```diff
@@ -0,0 +1,853 @@
+            ? html`
+                <span class="badge badge-pill badge-primary news-item-count">
+                  ${news_item_notification_count}
+                </span>
+              `
+            : ''}
+        </a>
+        <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
+          ${authn_is_administrator
+            ? html`
+                <button type="button" class="dropdown-item" id="navbar-administrator-toggle">
+                  ${access_as_administrator
+                    ? 'Turn off administrator access'
+                    : 'Turn on administrator access'}
+                </button>
+
+                <div class="dropdown-divider"></div>
+              `
+            : ''}
+          ${ViewTypeMenu({ resLocals })} ${AuthnOverrides({ resLocals, navbarType })}
+          ${authz_data?.mode === 'Exam'
+            ? html`
+                <div class="dropdown-item-text">
+                  Exam mode means you have a checked-in reservation to CBTF. Your interactions with
+                  PrairieLearn are limited to your checked-in exam for the duration of your
+                  reservation.
+                </div>
+                <div class="dropdown-divider"></div>
+              `
+            : ''}
+          ${!authz_data || authz_data?.mode !== 'Exam'
+            ? html`
+                <a class="dropdown-item" href="${config.urlPrefix}/request_course">
+                  Course Requests
+                </a>
+              `
+            : ''}
+          <a class="dropdown-item" href="${config.urlPrefix}/settings">Settings</a>
+          <a
+            class="dropdown-item news-item-link"
+            href="${urlPrefix}/news_items"
+            ${news_item_notification_count
+              ? html`
+                  title="News (${news_item_notification_count} unread)" aria-label="News
+                  (${news_item_notification_count} unread)"
+                `
+              : html`title="News" aria-label="News"`}
+          >
+            News
+            ${news_item_notification_count
+              ? html`
+                  <span class="badge badge-pill badge-primary news-item-link-count">
+                    ${news_item_notification_count}
+                  </span>
+                `
+              : ''}
+          </a>
+
+          <a class="dropdown-item" href="${config.urlPrefix}/logout">Log out</a>
+        </div>
+      </li>
+    </ul>
+  `;
+}
+
+function FlashMessages() {
+  const globalFlashColors = {
+    notice: 'info',
+    success: 'success',
+    warning: 'warning',
+    error: 'danger',
+  } as const;
+
+  return html`
+    <div class="mb-3">
+      ${flash(Object.keys(globalFlashColors) as FlashMessageType[]).map(
+        ({ type, message }) => html`
+          <div
+            class="alert alert-${globalFlashColors[
+              type
+            ]} border-left-0 border-right-0 rounded-0 mt-0 mb-0 alert-dismissible fade show js-global-flash"
```

Comment:
AFAICT the `js-global-flash` class is unused. Consider removing it now that this is being cleaned up.